### PR TITLE
SALTO-6516: (Salesforce) EmailTemplate with multiple attachments with the same name bug-fix

### PR DESCRIPTION
**Pull request by tamtamirr**
(Salesforce) EmailTemplate with multiple attachments with the same name bug-fix

---

Workspace diff: https://github.com/salto-io/tamir-sf/commit/647a9350892d91b00eeaade3a2939e7563b23007

Fixed an issue where the adapter would drop some static files and cause weird behavior on restore when EmailTemplate has attachments with the same name. 

---
_Release Notes_: 
_Salesforce Adapter_:
- Fixed an issue where the adapter would drop some static files and cause weird behavior on restore when EmailTemplate has attachments with the same name. Requires the `indexedEmailTemplateAttachments` optional feature.

---
_User Notifications_: 
_Salesforce_:
- Expect to see changes to existing EmailTemplate nacls and their static files when enabling the feature `indexedEmailTemplateAttachments`

### CODE REVIEW

**Review by amitgoldarb — APPROVED** [review]
LGTM
Let's let  @tomermevorach / @adamro  take a look at this before releasing.

File: packages/salesforce-adapter/src/filters/email_template_static_files.ts [hunk 1]
```diff
@@ -69,11 +70,11 @@ const organizeStaticFiles = async (instance: InstanceElement): Promise<void> =>
     try {
       instance.value.attachments = makeArray(instance.value.attachments)
       if (isEmailAttachmentsArray(instance.value.attachments)) {
-        instance.value.attachments.forEach(attachment => {
+        instance.value.attachments.forEach((attachment, index) => {
           attachment.content = createStaticFile(
             // attachment.content type is a string before the creation of the static file
             folderPath,
-            attachment.name,
+            fetchProfile.isFeatureEnabled('indexedEmailTemplateAttachments') ? `attachment${index}` : attachment.name,
```
[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/email_template_static_files.ts:77]: Makes sense to me

[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/email_template_static_files.ts:77]: Here's workspace diff after this change: https://github.com/salto-io/tamir-sf/commit/2d971cd337e45a5986e0526fc6fe16fbcc8679e6

[the remark this answers is not in the document]
Comment (tomermevorach) [reply packages/salesforce-adapter/src/filters/email_template_static_files.ts]: I think the name should be: index_attachmentName

Other than that, looks very nice

**Pull request merged**